shared/src/metabase/mbql/util.cljc

```diff
@@ -462,7 +462,10 @@
   "Is a given artihmetics clause operating on datetimes?"
   [clause]
   (mbql.match/match-one clause
-    #{:interval :relative-datetime}
+    #{:datetime-diff :temporal-extract}
+    false
+
+    #{:interval :relative-datetime :datetime-add}
```

Comment:
could you please add `datetime-subtract` to this set as well?

Reply:
@dpsutton @qnkhuat I discovered some issues with this `datetime-arithmetics?` function, and I think this implementation complicates them further. I've fixed the issues separately in [this PR](https://github.com/metabase/metabase/pull/26279). If approved, we can merge those changes into this PR.